File: cz/mg/vulkan/PFNvkGetDisplayPlaneCapabilitiesKHR.java
package cz.mg.vulkan;

public class PFNvkGetDisplayPlaneCapabilitiesKHR extends VkFunctionPointer {
    public PFNvkGetDisplayPlaneCapabilitiesKHR() {
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkGetDisplayPlaneCapabilitiesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetDisplayPlaneCapabilitiesKHR"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkDisplayModeKHR mode, VkUInt32 planeIndex, VkDisplayPlaneCapabilitiesKHR pCapabilities, VkResult rval){
        call(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.getSinkAddress(), mode != null ? mode.getVkAddress() : VkPointer.getSinkAddress(), planeIndex != null ? planeIndex.getVkAddress() : VkPointer.getSinkAddress(), pCapabilities != null ? pCapabilities.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddress());
    }

    private static native void call(long vkaddress, long physicalDevice, long mode, long planeIndex, long pCapabilities, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetDisplayPlaneCapabilitiesKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetDisplayPlaneCapabilitiesKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong mode, jlong planeIndex, jlong pCapabilities, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDisplayPlaneCapabilitiesKHR f = (PFN_vkGetDisplayPlaneCapabilitiesKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        *((VkDisplayModeKHR*)jniLongToPointer(mode)),
        *((uint32_t*)jniLongToPointer(planeIndex)),
        ((VkDisplayPlaneCapabilitiesKHR*)jniLongToPointer(pCapabilities))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetDisplayPlaneCapabilitiesKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong mode, jint planeIndex, jlong pCapabilities){
    (void)env;
    (void)jc;
    PFN_vkGetDisplayPlaneCapabilitiesKHR f = (PFN_vkGetDisplayPlaneCapabilitiesKHR)jniLongToFunctionPointer(address);
    return f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        *((VkDisplayModeKHR*)jniLongToPointer(mode)),
        (uint32_t)planeIndex,
        ((VkDisplayPlaneCapabilitiesKHR*)jniLongToPointer(pCapabilities))
    );
}

